### PR TITLE
Milestone 2: Add reducers and actions.

### PR DESCRIPTION
**Pull request by dfsalam**
# In this pull request I:
- ### Added Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- ### Created a directory that will contain all your Redux logic (/src/redux)
- ### Configured a Redux store (/src/redux/store.js)
### Defined a slice of state for books that:
- #### Includes an array of books (initial state: empty array)
- #### Includes a reducer that adds a book
- #### Includes a reducer that removes a book
### Defined a slice of state for categories that:
- #### Includes an array of categories (initial state: empty array)
- #### Includes a reducer that checks the status and always returns "Under construction" (the initial state should change to that string)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state) => {
```
Comment (t-yanick) [inline src/redux/books/booksSlice.js]: - [x] Your `addBook` reducer should have both a `state` and an `action`. At the moment, it is not having an `action`. Also, it not very necessary to initialize the properties of the book with initial values. Get a better view from the code snippet below:
```
addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },

```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      const book = {
+        item_id: 'item1', title: 'Python Data Analytics', author: 'Fabio Nelli', category: 'Python',
```
Comment (t-yanick) [inline src/redux/books/booksSlice.js]: - [x] `category` is not a necesssary property when adding a book at this time

File: src/redux/categories/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+const initialState = {
+  categories: [],
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => ({ ...state, categories: 'Under construction' }),
```
Comment (t-yanick) [inline src/redux/categories/categoriesSlice.js]: - [x] In the same light, your `checkStatus` reducer should have both a state and an action.
```javascript
checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi dfsalam , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  File structure follows ducks pattern
:heavy_check_mark: Added all reducers per project requirements

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] Great job working on this milestone. The live demo link in your README redirects to an error 404 page. If you don't yet have a live link, you can use the text `Coming Soon ...`. It is not professional to keep broken links in our codebase
Screen Shot 2023-03-01 at 8 59 59 PM


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @dfsalam,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Professional readme file ✔️
- No linter errors ✔️
- Good commit history ✔️
209362895-9762f2f8-13cf-4ce0-90b6-a672c4f04196

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**